lib/ansible/modules/crypto/luks_device.py

```diff
@@ -590,6 +671,12 @@ def run_module():
     if module.params['label'] is not None and module.params['type'] == 'luks1':
         module.fail_json(msg='You cannot combine type luks1 with the label option.')
 
+    if ((module.params['keyfile'] and module.params['passphrase'])
```

Comment:
Instead of this `if`, you should add `mutually_exclusive=[('keyfile', 'passprase'), ('new_keyfile', 'new_passphrase'), ('remove_keyfile', 'remove_passphrase')]` to the `AnsibleModule()` constructor call. (With appropriate code formatting if needed.)